drivers/at86rf215/at86rf215.c

```diff
@@ -0,0 +1,342 @@
+        if (is_subGHz(dev)) {
+            at86rf215_reg_write(dev, RG_BBC1_IRQM, 0);
+            at86rf215_reg_write(dev, RG_RF24_IRQM, 0);
+            at86rf215_reg_write(dev, RG_RF24_CMD, CMD_RF_SLEEP);
+
+        /* disable sub-GHz IRQs if the interface is not enabled */
+        } else {
+            at86rf215_reg_write(dev, RG_BBC0_IRQM, 0);
+            at86rf215_reg_write(dev, RG_RF09_IRQM, 0);
+            at86rf215_reg_write(dev, RG_RF09_CMD, CMD_RF_SLEEP);
+        }
+    }
+
+    /* disable clock output */
+#if AT86RF215_USE_CLOCK_OUTPUT == 0
+    at86rf215_reg_write(dev, RG_RF_CLKO, 0);
+#endif
+
+    /* allow to configure board-specific trim */
+#ifdef AT86RF215_TRIM_VAL
+    at86rf215_reg_write(dev, RG_RF_XOC, AT86RF215_TRIM_VAL | XOC_FS_MASK);
+#endif
+
+    /* enable TXFE & RXFE IRQ */
+    at86rf215_reg_write(dev, dev->BBC->RG_IRQM, BB_IRQ_TXFE | BB_IRQ_RXFE);
+
+    /* enable EDC IRQ */
+    at86rf215_reg_write(dev, dev->RF->RG_IRQM, RF_IRQ_EDC | RF_IRQ_TRXRDY);
+
+    /* set energy detect threshold to -84 dBm */
+    at86rf215_reg_write(dev, dev->BBC->RG_AMEDT, AT86RF215_EDT_DEFAULT);
+
+    /* enable address filter 0 */
+    at86rf215_reg_write(dev, dev->BBC->RG_AFC0, AFC0_AFEN0_MASK );
+    at86rf215_reg_write(dev, dev->BBC->RG_AMAACKPD, AMAACKPD_PD0_MASK);
+
+    /* enable auto-ACK with Frame Checksum & Data Rate derived from RX frame */
+    reg = AMCS_AACKFA_MASK | AMCS_AACKDR_MASK;
+    if (dev->flags & AT86RF215_OPT_AUTOACK) {
+        reg |= AMCS_AACK_MASK;
+    }
+
+    at86rf215_reg_write(dev, dev->BBC->RG_AMCS, reg);
+
+    /* set compatibility with first-gen 802.15.4 devices */
+    at86rf215_configure_legacy_OQPSK(dev, 0);
+
+    /* set default channel */
+    at86rf215_set_chan(dev, dev->netdev.chan);
+
+    /* set short and long address */
+    uint64_t long_addr;
+    memcpy(&long_addr, dev->netdev.long_addr, sizeof(long_addr));
+    at86rf215_set_addr_long(dev, long_addr);
+    at86rf215_set_addr_short(dev, 0, unaligned_get_u16(dev->netdev.short_addr));
+
+    /* set default PAN id */
+    at86rf215_set_pan(dev, 0, dev->netdev.pan);
+
+    /* set default TX power */
+    at86rf215_set_txpower(dev, AT86RF215_DEFAULT_TXPOWER);
+
+    /* start listening for incoming packets */
+    at86rf215_rf_cmd(dev, CMD_RF_RX);
+    at86rf215_await_state(dev, RF_STATE_RX);
+
+    dev->state = AT86RF215_STATE_IDLE;
+}
+
+ssize_t at86rf215_send(at86rf215_t *dev, const void *data, size_t len)
+{
+    /* check data length */
+    if (len > AT86RF215_MAX_PKT_LENGTH) {
+        DEBUG("[at86rf215] Error: data to send exceeds max packet size\n");
+        return -EOVERFLOW;
+    }
+
+    if (at86rf215_tx_prepare(dev)) {
+        return -EBUSY;
+    }
+
+    at86rf215_tx_load(dev, data, len, 0);
+    at86rf215_tx_exec(dev);
+    return len;
+}
+
+void at86rf215_tx_done(at86rf215_t *dev)
+{
+    uint8_t amcs = at86rf215_reg_read(dev, dev->BBC->RG_AMCS);
+
+    /* enable AACK, disable TX2RX */
```

Comment:
why does AACK need to be enabled here?

Reply:
It gets re-enabled if It was enabled before (because the radio is now in RX mode again and should send ACKs when receiving a frame) - I changed the comment to *re-enable*.